diff --git qxtglobalshortcut_p.h qxtglobalshortcut_p.h
--- qxtglobalshortcut_p.h
+++ qxtglobalshortcut_p.h
@@ -65,7 +65,11 @@ public:
     static QAbstractEventDispatcher::EventFilter prevEventFilter;
     static bool eventFilter(void* message);
 #else
+#  if QT_VERSION >= QT_VERSION_CHECK(6,0,0)
+    virtual bool nativeEventFilter(const QByteArray & eventType, void * message, qintptr * result) override;
+#  else
     virtual bool nativeEventFilter(const QByteArray & eventType, void * message, long * result);
+#  endif
 #endif // QT_VERSION < QT_VERSION_CHECK(5,0,0)
 #endif // Q_OS_MAC
 
diff --git qxtglobalshortcut_x11.cpp qxtglobalshortcut_x11.cpp
--- qxtglobalshortcut_x11.cpp
+++ qxtglobalshortcut_x11.cpp
@@ -34,7 +34,9 @@
 #   include <QX11Info>
 #else
 #   include <QApplication>
-#   include <qpa/qplatformnativeinterface.h>
+#   if QT_VERSION < QT_VERSION_CHECK(6,0,0)
+#       include <qpa/qplatformnativeinterface.h>
+#   endif
 #   include <xcb/xcb.h>
 #endif
 #include <X11/Xlib.h>
@@ -95,11 +97,17 @@ public:
     {
 #if QT_VERSION < QT_VERSION_CHECK(5,0,0)
         m_display = QX11Info::display();
-#else
+#elif QT_VERSION < QT_VERSION_CHECK(6,0,0)
         QPlatformNativeInterface *native = qApp->platformNativeInterface();
         void *display = native->nativeResourceForScreen(QByteArray("display"),
                                                         QGuiApplication::primaryScreen());
         m_display = reinterpret_cast<Display *>(display);
+#else
+        auto native = qApp->nativeInterface<QNativeInterface::QX11Application>();
+        if (native)
+            m_display = native->display();
+        else
+            m_display = nullptr;
 #endif
     }
 
@@ -164,7 +172,12 @@ bool QxtGlobalShortcutPrivate::eventFilter(void *message)
         unsigned int keystate = key->state;
 #else
 bool QxtGlobalShortcutPrivate::nativeEventFilter(const QByteArray & eventType,
-    void *message, long *result)
+    void *message,
+#if QT_VERSION >= QT_VERSION_CHECK(6,0,0)
+    qintptr *result)
+#else
+    long *result)
+#endif
 {
     Q_UNUSED(result);
 
diff --git qxtglobalshortcut_win.cpp qxtglobalshortcut_win.cpp
--- qxtglobalshortcut_win.cpp
+++ qxtglobalshortcut_win.cpp
@@ -37,7 +37,12 @@ bool QxtGlobalShortcutPrivate::eventFilter(void* message)
 {
 #else
 bool QxtGlobalShortcutPrivate::nativeEventFilter(const QByteArray & eventType,
-    void * message, long * result)
+    void * message,
+#if QT_VERSION >= QT_VERSION_CHECK(6,0,0)
+    qintptr * result)
+#else
+    long * result)
+#endif
 {
     Q_UNUSED(eventType);
     Q_UNUSED(result);
